gcc/rust/rustspec.cc

```diff
@@ -28,24 +28,10 @@ along with GCC; see the file COPYING3.  If not see
 
 /* This bit is set if we saw a `-xfoo' language specification.  */
 #define LANGSPEC (1 << 1)
-/* This bit is set if they did `-lm' or `-lmath'.  */
-#define MATHLIB (1 << 2)
-/* This bit is set if they did `-lpthread'.  */
-#define THREADLIB (1 << 3)
 /* This bit is set if they did `-lc'.  */
-#define WITHLIBC (1 << 4)
+#define WITHLIBC (1 << 2)
```

Comment:
Is this necessary? Users of rustc normally don't use libc directly, but link to libc because of the `#[link(name = "c")]` inside the libc bindings rust crate (`liblibc.rlib`), which this bit won't indicate.

Reply:
This mask is used to detect if the compiler was invoked with -lc so there is a variable flag saw_libc which can do some more magic with the linking options. Good catch, I had to go double-check the code myself after this comment.

I think this serves as a great start to strip the driver back which will help a lot in the future.

Reply:
I think a lot of this file is not compatible with how rust is expected to work in terms of linking, but that's fine because the ways used by rustc are not usable yet. For example, we shouldn't link with the standard library if a file contains `#![no_std]`. And as bjorn said, we should link with C libraries if they are mentioned in a [`link` attribute](https://doc.rust-lang.org/reference/items/external-blocks.html#the-link-attribute), which will again happen in source code and not from the compiler's invocation level. So I think a lot of the "link" options in that file will need to be reworked anyway once procedural macros are supported

Reply:
I mean specifically regarding to linking libc and librust, as the rustc compiler can also be invoked with `-W,l...` for linking libraries of course. But those aren't treated "differently" as those two in the rust-spec